x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -734,16 +735,9 @@ private PhysicalOperation planProject(ProjectExec project, LocalExecutionPlanner
         List<Integer> projectionList = new ArrayList<>(projections.size());
 
         Layout.Builder layout = new Layout.Builder();
-        Map<Integer, Layout.ChannelSet> inputChannelToOutputIds = new HashMap<>();
-        for (int index = 0, size = projections.size(); index < size; index++) {
-            NamedExpression ne = projections.get(index);
-
-            NameId inputId = null;
-            if (ne instanceof Alias a) {
-                inputId = ((NamedExpression) a.child()).id();
-            } else {
-                inputId = ne.id();
-            }
+        Map<Integer, Layout.ChannelSet> inputChannelToOutputIds = Maps.newHashMapWithExpectedSize(projections.size());
```

Comment:
We aren't using this collection from what I see? We're just doing a .get(), but never adding things

Reply:
Good catch, why do we even spend time and resources populating it then?

Reply:
I wonder, since that map is not used, does it mean we might duplicate actual data in the actual layout?

Reply:
Nice catch!

I think this map is unused for forever now, and people found this redundant map multiple times. I was sure one of the times, we actually had merged the fix...

I'm not sure how the channel sets were intended to be used for project, but the current project operator does [shallow copies of blocks](https://github.com/elastic/elasticsearch/blob/6e4cb8142b9e51ad2be5c32b5daaa575585fd02d/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java#L286), and this works well and is intended. It didn't always do this, we had to first make it work by making blocks refcounted. Maybe before that, the channel sets were supposed to help with deduplication.